### ISSUE TITLE
External force parameter to ForwardDynamics doesn't seem to influence the ODE

### ISSUE THREAD

**Issue by nashmit** [pyomeca/biorbd#265]
Hi,

I am trying to use the integration of RBDL ForwardDynamics from biorbd to compute sensitivity analyses wrt to external forces. This should be straightforward but it seems that applying external forces/torques has no effect.

Take, as an example, the pendulum.bioMod

`
from biorbd import *
#from casadi import *
#import biorbd
import numpy as np

model = biorbd.Model('finalProj_input/pendulum.bioMod')

Q = np.ones(model.nbQ())/10
QDot = np.ones(model.nbQ())/10
states = model.stateSet()
for state in states:
    state.setActivation(0.5)

joint_torque = model.muscularJointTorque(states, Q, QDot)

external_force_vector = np.zeros(6) # Spatial vector has 6 components
f_ext = biorbd.VecBiorbdSpatialVector()
f_ext.append(biorbd.SpatialVector(external_force_vector))

print('Spatial Vector:',f_ext[0].to_array())
QDDot = model.ForwardDynamics(Q, QDot, joint_torque,f_ext)
print(QDDot.to_array())

external_force_vector = np.array([1,2,3,4,5,6])
f_ext[0]=biorbd.SpatialVector(external_force_vector)
print('Spatial Vector:',f_ext[0].to_array())
QDDot = model.ForwardDynamics(Q, QDot, joint_torque,f_ext)
print(QDDot.to_array())
`

The output doesn't change:

`
Spatial Vector: [0. 0. 0. 0. 0. 0.]
[ 22.99403182 -24.39344599]
Spatial Vector: [1. 2. 3. 4. 5. 6.]
[ 22.99403182 -24.39344599]
`

Is there something wrong with my implementation or is this a bug?
I've used this as an example: https://github.com/pyomeca/bioptim/blob/master/bioptim/examples/torque_driven_ocp/spring_load.py

**Comment by pariterre** [comment]
Hi @nashmit !
My guess is that you did not included the tag `externarlforceindex` in you bioMod file to inform biorbd which segment the external forces are applied on.
There is an example of such tag in the bioMod in this file:
https://github.com/pyomeca/biorbd/blob/master/test/models/cubeWithExternalForces.bioMod

> What if the external 'force' is not applied directly to COM? ( I suppose the API covers this case too, or do I have to take care of it manually? )

There are utilities in Python to help for that, @Ipuch can you point any function that could help? :)

**Comment by nashmit** [comment]
> Hi @nashmit ! My guess is that you did not included the tag `externarlforceindex` in you bioMod file to inform biorbd which segment the external forces are applied on. There is an example of such tag in the bioMod in this file: https://github.com/pyomeca/biorbd/blob/master/test/models/cubeWithExternalForces.bioMod
> 
> > What if the external 'force' is not applied directly to COM? ( I suppose the API covers this case too, or do I have to take care of it manually? )
> 
> There are utilities in Python to help for that, @Ipuch can you point any function that could help? :)

That is true, I didn't add the tag `externarlforceindex` in bioMod file. I thought you are following the same design as RBDL and having a 1 to 1 vector component of external forces corresponding to each link in the chain but I see now that is not the case. (I thought, you are using this labeling only as a binding system for your optimal control implementation ) 

Thanks for the clarification!

**Comment by nashmit** [comment]
> Hi,
> 
> Until now, you have to manually express your spatial vector at the origin of the world coordinate system. Some lines do it in biorbd, but no method is available to do it automatically. Here is the example in c++ :
> 
> https://github.com/pyomeca/biorbd/blob/4a527e8c1959bafbe2f62149aa5b87b62b3014bb/src/RigidBody/Contacts.cpp#L479-L500
> 
> It could be a nice feature to add to the API.

Thanks for the help! :)

I can try to do it... if you can provide me with some assistance :)
